### DISCUSSION TITLE
The frist argument of useScroll  can support dom selector

### DISCUSSION THREAD

**Discussion by whateveryoudo** [vueuse/vueuse#4508] [Q&A]
### Clear and concise description of the problem

I find it not work when i use `documentByxx` as the first argument,  it works in onMounted, but I can't use the result in my template，I find use watch Api with `flush: post`, maybe dom element not a reactive element, get null ,  In addition， a lot of components ref.value not the element, maybe $el, I think provide a function to get the elem or compatible  other circumstances

### Suggested solution

start listener in nexttick or  use onmounted inside

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by OrbisK** [comment]
`documentByxx` is most likly not there when calling it before mount. In most cases you can use [`useTemplateRef()`](https://vuejs.org/guide/essentials/template-refs.html#template-refs) to get an element. 

Why do you need `documentBy`?
